lib/datasource/maven/index.ts

```diff
@@ -27,20 +28,49 @@ export const defaultRegistryUrls = [MAVEN_REPO];
 export const defaultVersioning = mavenVersioning.id;
 export const registryStrategy = 'merge';
 
+const mavenCentralHtmlVersionRegex =
+  /<a href="(?<version>[^"]+)\/" title="\k<version>\/">\k<version>\/<\/a>\s+(?:(?<releaseTimestamp>\d\d\d\d-\d\d-\d\d \d\d:\d\d)|-)\s+-/gi;
```

Comment:
RE2 can't handle it :(

Reply:
Maybe we should split the string instead of regex, or in advance of regex?

Reply:
I actually don't think this regex is necessary slow. The `<a href="···/"` matches only directories, and we exclude `../` path early by requiring `title` and content to match captured backreference. Once done, it will match only versions or will short-circuit earlier instead.

Reply:
i think it's because of backtracking, which is explicit forbidden `\k<version>`